src/helpers/components/Highlight.tsx

```diff
@@ -0,0 +1,48 @@
+/** @jsx h */
+import { h } from 'preact';
+
+import { Highlight as HighlightUiComponent } from '../../components/Highlight/Highlight';
+import getHighlightedParts from '../../lib/utils/getHighlightedParts';
+import getPropertyByPath from '../../lib/utils/getPropertyByPath';
+import unescape from '../../lib/utils/unescape';
+import { warning } from '../../lib/utils/logger';
+
+import type { BaseHit, Hit, PartialKeys } from '../../types';
+import type { HighlightProps as HighlightUiComponentProps } from '../../components/Highlight/Highlight';
+
+export type HighlightProps<THit extends Hit<BaseHit>> = {
+  hit: THit;
+  attribute: keyof THit | string[];
+  cssClasses?: HighlightUiComponentProps['classNames'];
+} & PartialKeys<
+  Omit<HighlightUiComponentProps, 'parts' | 'classNames'>,
+  'highlightedTagName' | 'nonHighlightedTagName' | 'separator'
+>;
+
+export function Highlight<THit extends Hit<BaseHit>>({
+  hit,
+  attribute,
+  cssClasses,
+  ...props
+}: HighlightProps<THit>) {
+  const property =
+    getPropertyByPath(hit._highlightResult, attribute as string) || [];
+  const properties = Array.isArray(property) ? property : [property];
```

Comment:
not sure if now is the moment, but on Angular InstantSearch (I think) we fall back to the attribute value instead of the highlighted one, which means that if something is sometimes a number, you don't need to add a conditional to the code. Maybe this could be done with an option, not by default? maybe later?

Reply:
Yes I saw the comment in the existing components and hesitated to implement because I didn't know how much we've discussed and agreed to this pattern and considered alternatives. Also, this would be a different behavior from the existing components (although not breaking) so I'm not sure we should do it now.